File: sample/src/main/java/tech/thdev/androidrecyclerview/adapter/header_footer/holder/CustomScrollSmallImageViewHolder.java
package tech.thdev.androidrecyclerview.adapter.header_footer.holder;

import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import butterknife.BindView;
import tech.thdev.androidrecyclerview.R;
import tech.thdev.androidrecyclerview.data.LocalImage;
import tech.thdev.support.widget.adapter.simple.holder.BaseViewHolder;

/**
 * Created by Tae-hwan on 01/11/2016.
 */

public class CustomScrollSmallImageViewHolder extends BaseViewHolder<LocalImage> {

    @BindView(R.id.image_view)
    ImageView imageView;

    @BindView(R.id.tv_message)
    TextView tvMessage;

    public CustomScrollSmallImageViewHolder(@Nullable ViewGroup parent, @NotNull RecyclerView.Adapter<?> adapter) {
        super(R.layout.item_small_image, parent, adapter);
    }

    @Override
    public void onViewHolder(@Nullable LocalImage localImage, int position) {
        if (localImage == null) return;

        tvMessage.setText(localImage.getTitle());

        Glide.with(getContext())
                .load(localImage.getResource())
                .centerCrop()
                .placeholder(0)
                .into(imageView);
    }
}

File: sample/src/main/java/tech/thdev/androidrecyclerview/ui/header_footer/custom/adapter/holder/CustomScrollSmallImageViewHolder.kt
package tech.thdev.androidrecyclerview.ui.header_footer.custom.adapter.holder

import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.item_large_view.view.tv_message
import kotlinx.android.synthetic.main.item_small_image.view.*
import tech.thdev.androidrecyclerview.R
import tech.thdev.androidrecyclerview.data.LocalImage
import tech.thdev.support.widget.adapter.simple.holder.BaseViewHolder

/**
 * Created by Tae-hwan on 01/11/2016.
 */
class CustomScrollSmallImageViewHolder(
    parent: ViewGroup,
    adapter: RecyclerView.Adapter<*>
) : BaseViewHolder<LocalImage?>(R.layout.item_small_image, parent, adapter) {

    override fun onBindViewHolder(item: LocalImage?, position: Int) {
        if (item == null) return
        itemView.tv_message.text = item.title
        Glide.with(context)
            .load(item.resource)
            .centerCrop()
            .into(itemView.image_view)
    }
}